cg/meta/archive/ddn_dataflow.py

```diff
@@ -186,34 +186,39 @@ def auth_header(self) -> Dict[str, str]:
             self._refresh_auth_token()
         return {"Authorization": f"Bearer {self.auth_token}"}
 
-    def archive_folders(self, sources_and_destinations: Dict[Path, Path]) -> bool:
-        """Archives all folders provided, to their corresponding destination, as given by sources and destination parameter."""
-        transfer_data: List[TransferData] = [
-            TransferData(source=source.as_posix(), destination=destination.as_posix())
-            for source, destination in sources_and_destinations.items()
-        ]
-        transfer_request: TransferPayload = TransferPayload(files_to_transfer=transfer_data)
+    def archive_folders(self, sources_and_destinations: List[TransferData]) -> int:
+        """Archives all folders provided, to their corresponding destination,
+        as given by sources and destination parameter."""
+        transfer_request: TransferPayload = TransferPayload(
+            files_to_transfer=sources_and_destinations
+        )
         transfer_request.trim_paths(attribute_to_trim=SOURCE_ATTRIBUTE)
         transfer_request.add_repositories(
             source_prefix=self.local_storage, destination_prefix=self.archive_repository
         )
-        return transfer_request.post_request(
+        response: Response = transfer_request.post_request(
             headers=dict(self.headers, **self.auth_header),
             url=urljoin(base=self.url, url=DataflowEndpoints.ARCHIVE_FILES),
         )
+        if response.ok:
+            return response.json()["job_id"]
+        else:
+            raise IOError(response.text)
 
-    def retrieve_folders(self, sources_and_destinations: Dict[Path, Path]) -> bool:
+    def retrieve_folders(self, sources_and_destinations: List[TransferData]) -> int:
         """Retrieves all folders provided, to their corresponding destination, as given by the sources and destination parameter."""
-        transfer_data: List[TransferData] = [
-            TransferData(source=source.as_posix(), destination=destination.as_posix())
-            for source, destination in sources_and_destinations.items()
-        ]
-        transfer_request: TransferPayload = TransferPayload(files_to_transfer=transfer_data)
+        transfer_request: TransferPayload = TransferPayload(
```

Comment:
Consider extracting a separate method for creating the `transfer_request` here. Something like:
```python
def retrieve_folders(self, sources_and_destinations: List[TransferData]) -> int:
    transfer_request: TransferRequest = self.create_transfer_request(sources_and_destinations)
     
     response: Response = transfer_request.post_request(
         headers=dict(self.headers, **self.auth_header),
         url=urljoin(base=self.url, url=DataflowEndpoints.RETRIEVE_FILES),
      )
        
      response.raise_for_status()
      return response.json()["job_id"]
```

Reply:
The `response.raise_for_status()` could also be moved into the method sending the request, that might be even cleaner. That way we won't have to raise the exceptions everywhere

Reply:
Moved `response.raise_for_status()` into `post_request`. Will look inte extracting the creation into a method